grapher/stackedCharts/MarimekkoChart.tsx

```diff
@@ -0,0 +1,1539 @@
+import * as React from "react"
+import {
+    min,
+    max,
+    maxBy,
+    last,
+    flatten,
+    excludeUndefined,
+    sortBy,
+    sumBy,
+    sum,
+    minBy,
+} from "../../clientUtils/Util"
+import { action, computed, observable } from "mobx"
+import { observer } from "mobx-react"
+import { Bounds, DEFAULT_BOUNDS } from "../../clientUtils/Bounds"
+import {
+    BASE_FONT_SIZE,
+    EntitySelectionMode,
+    SeriesName,
+} from "../core/GrapherConstants"
+import { DualAxisComponent } from "../axis/AxisViews"
+import { NoDataModal } from "../noDataModal/NoDataModal"
+import { AxisConfig } from "../axis/AxisConfig"
+import { ChartInterface } from "../chart/ChartInterface"
+import { OwidTable } from "../../coreTable/OwidTable"
+import { autoDetectYColumnSlugs, makeSelectionArray } from "../chart/ChartUtils"
+import { stackSeries } from "./StackedUtils"
+import { ChartManager } from "../chart/ChartManager"
+import { Color, Time } from "../../clientUtils/owidTypes"
+import { StackedPoint, StackedSeries } from "./StackedConstants"
+import { ColorSchemes } from "../color/ColorSchemes"
+import {
+    EntityId,
+    EntityName,
+    LegacyOwidRow,
+    OwidTableSlugs,
+} from "../../coreTable/OwidTableConstants"
+import {
+    LegendAlign,
+    HorizontalCategoricalColorLegend,
+    HorizontalColorLegendManager,
+} from "../horizontalColorLegend/HorizontalColorLegends"
+import { CategoricalBin } from "../color/ColorScaleBin"
+import { CoreColumn } from "../../coreTable/CoreTableColumns"
+import { TippyIfInteractive } from "../chart/Tippy"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faInfoCircle } from "@fortawesome/free-solid-svg-icons/faInfoCircle"
+import { DualAxis, HorizontalAxis, VerticalAxis } from "../axis/Axis"
+import { ColorScale, ColorScaleManager } from "../color/ColorScale"
+import {
+    ColorScaleConfig,
+    ColorScaleConfigDefaults,
+} from "../color/ColorScaleConfig"
+import { ColorSchemeName } from "../color/ColorConstants"
+import { color } from "d3-color"
+import { SelectionArray } from "../selection/SelectionArray"
+import { ColorScheme } from "../color/ColorScheme"
+import { CoreRow } from "../../coreTable/CoreTableConstants"
+import _ from "lodash"
+
+export interface MarimekkoChartManager extends ChartManager {
+    endTime?: Time
+    excludedEntities?: EntityId[]
+    matchingEntitiesOnly?: boolean
+}
+
+interface EntityColorData {
+    color: Color
+    colorDomainValue: string
+}
+// Points used on the X axis
+interface SimplePoint {
+    value: number
+    entity: string
+    time: number
+}
+
+export interface SimpleChartSeries {
+    seriesName: string
+    points: SimplePoint[]
+}
+
+enum BarShape {
+    Bar,
+    BarPlaceholder,
+}
+
+interface Bar {
+    kind: BarShape.Bar
+    color: Color // color from the variable
+    seriesName: string
+    yPoint: StackedPoint<EntityName>
+}
```

Comment:
I don't think that sharing this definition gains us much. I'd rather be able to call the point yPoint here than re-use the type. If it turns out later that we can share more code because StackedDiscreetBarChart and Marimekko get more similar then we can revisit but right now I think the gain is low and clarity would suffers somewhat